src/app/qgisapp.cpp

```diff
@@ -11815,6 +11880,16 @@ void QgisApp::keyPressEvent( QKeyEvent *e )
   }
 }
 
+void QgisApp::newProfile()
+{
+  QString text = QInputDialog::getText( this, tr( "New profile name" ), tr( "New profile name" ) );
+  if ( text.isEmpty() )
+    return;
+
+  userProfileManager()->createUserProfile( text );
```

Comment:
I think the returned error should be checked after this (and feedback given to the user if something went wrong)